packages/core-sdk/src/resources/group.ts

```diff
@@ -388,7 +395,78 @@ export class GroupClient {
       handleError(error, "Failed to register group and attach license and add ips");
     }
   }
+  /**
+   * Collect royalties for the entire group and distribute the rewards to each member IP's royalty vault.
+   *
+   * Emits an on-chain {@link https://github.com/storyprotocol/protocol-core-v1/blob/v1.3.1/contracts/interfaces/modules/grouping/IGroupingModule.sol#L38 | `CollectedRoyaltiesToGroupPool`} event.
+   */
+  public async collectAndDistributeGroupRoyalties({
+    groupIpId,
+    currencyTokens,
+    memberIpIds,
+    txOptions,
+  }: CollectAndDistributeGroupRoyaltiesRequest): Promise<CollectAndDistributeGroupRoyaltiesResponse> {
+    try {
+      const object: GroupingWorkflowsCollectRoyaltiesAndClaimRewardRequest = {
```

Comment:
`object` is not a very good variable name. I recommend something like `protocolParams` or something similar to be a bit more descriptive on what this is for.